src/gui/folderstatusmodel.cpp

```diff
@@ -1226,7 +1227,10 @@ bool FolderStatusModel::SubFolderInfo::hasLabel() const
 void FolderStatusModel::SubFolderInfo::resetSubs(FolderStatusModel *model, QModelIndex index)
 {
     _fetched = false;
-    delete _fetchingJob;
+    if (_fetchingJob) {
```

Comment:
Does this guarantee that no signals from the stale job will be emitted? Maybe disconnect them?